components/org.apache.stratos.common/src/main/java/org/apache/stratos/common/statistics/publisher/ThriftClientConfigParser.java

```diff
@@ -0,0 +1,117 @@
+ * @author  supunr
+ * @contact supunr@wso2.com
+ * @date    1/27/15
+ */
+public class ThriftClientConfigParser {
+    /**
+     * Fields to be read from the thrift-client-config.xml file
+     */
+    private static final String USERNAME_ELEMENT = "username";
+    private static final String PASSWORD_ELEMENT = "password";
+    private static final String IP_ELEMENT = "ip";
+    private static final String PORT_ELEMENT = "port";
+
+    /**
+     * This method reads thrift-client-config.xml file and assign necessary credential
+     * values into thriftClientInfo object.  A singleton design has been implemented
+     * with the use of thriftClientIConfig class.
+     * <p>
+     * The filePath argument is the path to thrift-client-config.xml file
+     *
+     * @param filePath the path to thrift-client-config.xml file
+     * @return         ThriftClientConfig object
+     */
+    public static ThriftClientConfig parse(String filePath) {
+        try {
+            ThriftClientConfig thriftClientIConfig = new ThriftClientConfig();
+            ThriftClientInfo thriftClientInfo = new ThriftClientInfo();
+            thriftClientIConfig.setThriftClientInfo(thriftClientInfo);
+
+            OMElement document = AxiomXpathParserUtil.parse(new File(filePath));
+            Iterator thriftClientIterator = document.getChildElements();
+
+            String userNameValuesStr = null;
+            String passwordValueStr = null;
+            String ipValuesStr = null;
+            String portValueStr = null;
+
+            // Iterate the thrift-client-config.xml file and read child element
+            // consists of credential information necessary for WSO2CEPStatisticsPublisher
+            while (thriftClientIterator.hasNext()) {
+                OMElement thriftClientElement = (OMElement) thriftClientIterator.next();
+
+                if (USERNAME_ELEMENT.equals(thriftClientElement.getQName().getLocalPart())) {
+                    userNameValuesStr = thriftClientElement.getText();
+                    thriftClientInfo.setUsername(userNameValuesStr);
+                }
+
+                if (PASSWORD_ELEMENT.equals(thriftClientElement.getQName().getLocalPart())) {
+                    passwordValueStr = thriftClientElement.getText();
+                    thriftClientInfo.setPassword(passwordValueStr);
+                }
+
+                if (IP_ELEMENT.equals(thriftClientElement.getQName().getLocalPart())) {
+                    ipValuesStr = thriftClientElement.getText();
+                    thriftClientInfo.setIp(ipValuesStr);
+                }
+
+                if (PORT_ELEMENT.equals(thriftClientElement.getQName().getLocalPart())) {
+                    portValueStr = thriftClientElement.getText();
+                    thriftClientInfo.setPort(portValueStr);
+                }
+            }
+
+            if (userNameValuesStr == null) {
+                throw new RuntimeException("Username value not found in thriftClientConfiguration");
```

Comment:
Shall we change the text "thriftClientConfiguration" to "thrift client configuration"?